virtcontainers/device/config/config.go

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2017-2018 Intel Corporation
+// Copyright (c) 2018 Huawei Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package config
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/go-ini/ini"
+)
+
+// DeviceType indicates device type
+type DeviceType string
+
+const (
+	// DeviceVFIO is the VFIO device type
+	DeviceVFIO DeviceType = "vfio"
+
+	// DeviceBlock is the block device type
+	DeviceBlock DeviceType = "block"
+
+	// DeviceGeneric is a generic device type
+	DeviceGeneric DeviceType = "generic"
+
+	//VhostUserSCSI - SCSI based vhost-user type
+	VhostUserSCSI = "vhost-user-scsi-pci"
+
+	//VhostUserNet - net based vhost-user type
+	VhostUserNet = "virtio-net-pci"
+
+	//VhostUserBlk represents a block vhostuser device type
+	VhostUserBlk = "vhost-user-blk-pci"
+)
+
+// Defining these as a variable instead of a const, to allow
+// overriding this in the tests.
```

Comment:
not sure if this comment was there in original code, but wanted to comment that I appreciate being explicit here on why you're not using a const! Thanks!

Reply:
Haha, this is from original code. But I also appreciate this, that's why I keep it after the refactor :-)